eplus.py

```diff
@@ -37,31 +39,127 @@ def _get_eplus_data(session, eplus_url):
     return result
 
 
+class EplusSessionUpdater(Thread):
+    """
+    Cookies for Eplus expire after about 1 hour.
+    To keep our live streaming going, we have to refresh them in time,
+    otherwise we may got HTTP 403 and no new stream could be downloaded.
+    """
+
+    def __init__(self, session, eplus_url):
+        self._eplus_url = eplus_url
+        self._session = session
+        self._closed = Event()
+        self._retries = 0
+        self._last_expire_timestamp = time.time()
+        self._log = logging.getLogger(f"{__name__}.{self.__class__.__qualname__}")
+
+        super().__init__(name=self.__class__.__qualname__, daemon=True)
+
+    def close(self):
+        if self._closed.is_set():
+            """
+            "close(self)" will be called multiple times during the cleanup process of Streamlink.
+            If Python is about to exit, logging something will raise an exception:
+            > ImportError: sys.meta_path is None, Python is likely shutting down <
+            """
```

Comment:
This comment probably isn't necessary, being able to call `close()` multiple times on an already closed object is pretty standard python behavior

Reply:
Well, although it's a standard behavior, the exception indeed surprised and confused me when I first see it... I learned it from streamlink/streamlink#3357 . Then, as a memo, I added the above comment.